q2_quality_control/decontam_assets/run_decontam.R

```diff
@@ -0,0 +1,96 @@
+#!/usr/bin/env Rscript
+
+library("decontam")
+library("optparse")
+
+cat(R.version$version.string, "\n")
+errQuit <- function(mesg, status=1) { message("Error: ", mesg); q(status=status) }
+
+option_list = list(
+  make_option(c("--asv_table_path"), action="store", default='NULL', type='character',
+              help="File path to table in .csv format "),
+  make_option(c("--threshold"), action="store", default='NULL', type='character',
+              help="threshold value for decontam algorithm"),
+  make_option(c("--decon_method"), action="store", default='NULL', type='character',
+              help="algorithm mode"),
+  make_option(c("--output_track"), action="store", default='NULL', type='character',
+              help="File path to tracking tsv file. If already exists, will be overwritten"),
+  make_option(c("--meta_table_path"), action="store", default='NULL', type='character',
+              help="File path to metadata in .tsv format"),
+  make_option(c("--freq_con_column"), action="store", default='NULL', type='character',
+              help="Name of column for frequency method"),
+  make_option(c("--prev_control_or_exp_sample_column"), action="store", default='NULL', type='character',
+              help="Name of column for prevalence method"),
+  make_option(c("--prev_control_sample_indicator"), action="store", default='NULL', type='character',
+              help="Indicator to identify control samples")
+)
+opt = parse_args(OptionParser(option_list=option_list))
+
+inp.loc <- opt$asv_table_path
+threshold <- if(opt$threshold=='NULL') NULL else as.numeric(opt$threshold)
+out.track <- opt$output_track
+metadata.loc<-opt$meta_table_path
+decon.mode<-opt$decon_method
+prev.control.col <- opt$prev_control_or_exp_sample_column
+prev.id.controls<-opt$prev_control_sample_indicator
+freq.control.col<-opt$freq_con_column
+
+if(!file.exists(inp.loc)) {
+  errQuit("Input ASV table does not exist.")
+}else if(!file.exists(metadata.loc)) {
+  errQuit("Input metadata file does not exist.")
+}else{
+  print("Input ASV and Metadata files found")
+}
+
+meta_data_cols <-function(metadata_df, control.col){
+  control_vec<-c()
+  index<-0
+  for (id in colnames(metadata_df)) {
+    index=index+1
+    if(tolower(id) == tolower(control.col)){
+      control_vec<-metadata_df[,c(index)]
+    }
+  }
+  return(control_vec)
+}
+
+outputer<-function(decon_output, out.track,asv_df, out.path){
+  cat("7) Write output\n")
+  write.table(decon_output, out.track, sep="\t",
+              row.names=TRUE, col.names=NA, quote=FALSE)
```

Comment:
Somewhere in this scheme the row.names are prefixed with an `X` when an ID starts with a number. Which while understandable, would cause the IDs to not match elsewhere.